lib/workload/stateful/stacks/shared/constructs/event-bus/custom-event-archiver/construct/universal-event-archiver/index.ts

```diff
@@ -0,0 +1,73 @@
+import path from 'path';
+import { IBucket } from 'aws-cdk-lib/aws-s3';
+import { LambdaFunction } from 'aws-cdk-lib/aws-events-targets';
+import { PolicyStatement } from 'aws-cdk-lib/aws-iam';
+import { aws_lambda, Duration, Stack } from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import { PythonFunction } from '@aws-cdk/aws-lambda-python-alpha';
+import { Architecture } from 'aws-cdk-lib/aws-lambda';
+import { IEventBus, Rule } from 'aws-cdk-lib/aws-events';
+import { IVpc, ISecurityGroup } from 'aws-cdk-lib/aws-ec2';
+
+export interface UniversalEventArchiverConstructProps {
+  vpc: IVpc;
+  lambdaSG: ISecurityGroup;
+  archiveBucket: IBucket;
+  eventBus: IEventBus;
+}
+
+export class UniversalEventArchiverConstruct extends Construct {
+  private readonly id: string;
+  private readonly lambdaRuntimePythonVersion: aws_lambda.Runtime = aws_lambda.Runtime.PYTHON_3_12;
+
+  constructor(scope: Construct, id: string, props: UniversalEventArchiverConstructProps) {
+    super(scope, id);
+
+    this.id = id;
+
+    const eventBus = props.eventBus;
+    const archiveBucket = props.archiveBucket;
+    const lambdaSG = props.lambdaSG;
+
+    const archiveEventFunction = new PythonFunction(this, 'UniversalEventArchiver', {
+      entry: path.join(__dirname, '../../archiver-service'),
+      runtime: this.lambdaRuntimePythonVersion,
+      environment: {
+        BUCKET_NAME: archiveBucket.bucketName,
+      },
+      vpc: props.vpc,
+      vpcSubnets: { subnets: props.vpc.privateSubnets },
+      securityGroups: [lambdaSG],
+      architecture: Architecture.ARM_64,
+      timeout: Duration.seconds(28),
+      index: 'universal_event_achiver.py',
```

Comment:
ditto fix filename:

```
index: 'universal_event_archiver.py',
```

Reply:
I thought it should be a noun, like "archiver" sth, but happy to have any suggestion

Reply:
Current filename is fine. It is just the typo.